### PR TITLE
Drop code that explicitly supported python 2.

### PR DESCRIPTION
**Pull request by sodul**
Tested with `green green` with python 3.6 (some of the test code uses f-strings so we cannot test with python 3.5).

### CODE REVIEW

**Comment by CleanCut** [comment]
It's the end of an era. 😢 

Time to move on! This is looking pretty good. ❤️

Some minor things to address:

- [ ] Fix the linting errors from `black`
- [ ] Update the changelog
- [ ] Update the readme

If you will enable the `Allow edits from maintainers` setting on this PR, I can fixup minor issues like these for you instead.

**Comment by sodul** [comment]
Sorry The `Allow edits from maintainers` option does not exist for me here.

image

**Comment by sodul** [comment]
I've run black 23.1.2 (latest), added Python 3.11 to the CI checks and updated the .md files.

**Comment by CleanCut** [comment]
> Sorry, the `Allow edits from maintainers` option does not exist for me here.

Huh. Weird. Do you see it in a place more like this?

image

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Unreleased
 #### Date TBD
 
+- Drop Python 2 support.
+- Run black 23.1.2 on the code to apply the newest style.
```
Comment (CleanCut) [inline CHANGELOG.md:5]: Resolving lint errors is implicit.

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Features
 - **Convenient** - Bash-completion and ZSH-completion of options and test targets.
 - **Thorough** - Built-in integration with [coverage](http://nedbatchelder.com/code/coverage/).
 - **Embedded** - Can be run with a setup command without in-site installation.
-- **Modern** - Supports Python 3.5+. Additionally, [PyPy](http://pypy.org) is supported on a best-effort basis.
+- **Modern** - Supports Python 3.5+. Additionally, [PyPy](http://pypy.org) is supported on a best-effort basis. Python 2 is no longer supported.
```
Comment (CleanCut) [inline README.md:19]: Hmm. This was already fine. I guess there's nothing to update in the readme after all.

```suggestion
- **Modern** - Supports Python 3.5+. Additionally, [PyPy](http://pypy.org) is supported on a best-effort basis.
```

**Comment by CleanCut** [comment]
> I've run black 23.3.0 (latest),

That's really odd, because you're commit has lots of whitespace changes that are showing up as _new_ linting errors, and when I run black locally, it reverts your changes. Do you, perhaps, have some different black configuration in your local environment or something?

**Comment by sodul** [comment]
Sorry, I really cannot find the checkbox to allow edits anywhere. The fork is tied to our work Org and might not give the option somehow.

I've added your suggestions to the PR.

As for black, I re-ran it from a vanilla image to remove the possibility of an other config on my machine. I hope this time it will be better.

We usually run `gray` on our codebase and not `black` as it is more flexible. We do have it call `black` for the general whitespace and line splitting handling, but we prefer single quotes for example.

**Comment by sodul** [comment]
@CleanCut is there anything else you want me to change here?

**Comment by CleanCut** [comment]
Sorry for the delay, I was away for several days. 😄 

This looks great!

**Pull request merged**